velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -235,23 +402,77 @@ void WindowFuzzer::go() {
     const auto call =
         makeFunctionCall(signature.name, argNames, false, false, ignoreNulls);
 
+    auto [frameType, startBoundType, endBoundType] =
+        frameWindowTypeAndBoundType();
+
+    std::vector<core::WindowNode::BoundType> kBoundOptions = {
+        core::WindowNode::BoundType::kPreceding,
+        core::WindowNode::BoundType::kFollowing};
+    bool isKRangeFrame = frameType == core::WindowNode::WindowType::kRange &&
+        (std::find(
+             kBoundOptions.begin(), kBoundOptions.end(), startBoundType) !=
+             kBoundOptions.end() ||
+         std::find(kBoundOptions.begin(), kBoundOptions.end(), endBoundType) !=
+             kBoundOptions.end());
+
     std::vector<SortingKeyAndOrder> sortingKeysAndOrders;
+    TypeKind orderByTypeKind;
+    std::string orderByKey = "";
     // 50% chance without order-by clause.
-    if (vectorFuzzer_.coinToss(0.5)) {
+    if (isKRangeFrame) {
+      // k-range frames need 1 order by key
+      sortingKeysAndOrders =
+          generateSortingKeysAndOrders("s", argNames, argTypes, isKRangeFrame);
+      orderByTypeKind = argTypes.back()->kind();
+      orderByKey = sortingKeysAndOrders[0].key_;
+    } else if (vectorFuzzer_.coinToss(0.5)) {
       sortingKeysAndOrders =
           generateSortingKeysAndOrders("s", argNames, argTypes);
     }
     const auto partitionKeys = generateSortingKeys("p", argNames, argTypes);
-    const auto [frameClause, isRowsFrame] = generateFrameClause();
-    const auto input = generateInputDataWithRowNumber(
-        argNames, argTypes, partitionKeys, signature);
+
     // If the function is order-dependent or uses "rows" frame, sort all input
     // rows by row_number additionally.
-    if (requireSortedInput || isRowsFrame) {
+    if ((requireSortedInput && !isKRangeFrame) ||
+        frameType == core::WindowNode::WindowType::kRows) {
```

Comment:
Could you please share why the ties need to be broken in the order by column in case of kRange frames? This column was removed for kRange frames since we want only one orderBy column, which is randomly generated with vectorFuzzer.

Reply:
Hi @kagamiori, thanks for pointing this out. I am planning to use the information about order dependent functions to decide what the order by column would be. When we have kRange frames: if the function is order dependent, `row_number` will be used as the order by column, otherwise the order by column will be generated randomly and `row_number` column will not be added. will work on the prototype, compare the results with Presto query runner and share the findings by today.